include/igraph_types.h

```diff
@@ -87,6 +88,9 @@ typedef int    igraph_bool_t;
 #  define IGRAPH_UINT_MIN UINT32_MIN
 #endif
 
+/* Largest positive value for igraph_real_t that can safely represent integers. */
+#define IGRAPH_MAX_EXACT_REAL ((double)(1LL << DBL_MANT_DIG))
+
```

Comment:
Let's put this into `safe_intop.h`. It does not need to be public.

Reply:
It seems like this is done now.